src/test/java/com/artipie/npm/NpmProxyITCase.java

```diff
@@ -0,0 +1,217 @@
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.artipie.npm;
+
+import com.artipie.ArtipieServer;
+import com.artipie.RepoConfigYaml;
+import com.artipie.RepoPerms;
+import com.artipie.asto.Key;
+import com.artipie.asto.Storage;
+import com.artipie.asto.fs.FileStorage;
+import com.artipie.asto.test.TestResource;
+import com.artipie.http.rs.RsStatus;
+import com.artipie.nuget.RandomFreePort;
+import com.artipie.test.RepositoryUrl;
+import com.artipie.test.TestContainer;
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.nio.file.Path;
+import java.util.Arrays;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.core.IsEqual;
+import org.hamcrest.text.StringContainsInOrder;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.EnabledOnOs;
+import org.junit.jupiter.api.condition.OS;
+import org.junit.jupiter.api.io.TempDir;
+
+/**
+ * Integration test for {@link com.artipie.npm.proxy.http.NpmProxySlice}.
+ * @since 0.13
+ * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
+ */
+@SuppressWarnings("PMD.AvoidDuplicateLiterals")
+@EnabledOnOs({OS.LINUX, OS.MAC})
+@Disabled
+final class NpmProxyITCase {
+
+    /**
+     * Project name.
+     */
+    private static final String PROJ = "@hello/simple-npm-project";
+
+    /**
+     * Origin repo name.
+     */
+    private static final String ORIGIN = "npm-origin";
+
+    /**
+     * Proxy repo name.
+     */
+    private static final String PROXY = "npm-proxy";
+
+    /**
+     * Temporary directory for all tests.
+     * @checkstyle VisibilityModifierCheck (3 lines)
+     */
+    @TempDir
+    Path tmp;
+
+    /**
+     * Test origin.
+     */
+    private ArtipieServer origin;
+
+    /**
+     * Test proxy.
+     */
+    private ArtipieServer proxy;
+
+    /**
+     * Container.
+     */
+    private TestContainer cntn;
+
+    @Test
+    void installFromProxy() throws Exception {
+        final boolean anonymous = true;
+        this.init(anonymous);
+        final HttpURLConnection con = (HttpURLConnection) new URL(
+            String.format(
+                // @checkstyle LineLengthCheck (1 line)
+                "http://localhost:%d/%s/@hello/simple-npm-project/-/@hello/simple-npm-project-1.0.1.tgz", this.origin.port(), NpmProxyITCase.ORIGIN
+            )
+        ).openConnection();
+        con.setRequestMethod("GET");
+        con.setDoOutput(true);
+        MatcherAssert.assertThat(
+            "Verify that tgz file exists. Response status is 200",
+            con.getResponseCode(),
+            new IsEqual<>(Integer.parseInt(RsStatus.OK.code()))
+        );
```

Comment:
@genryxy not sure why do we need this check. The test checks that `npm install` works later, this check does not look to be required

Reply:
@olegmoz thanks, I wanted to show by this check that the file can actually be obtained from origin repo. Probably I should remove this check or write about removing in the created issue related to enabling test.

Reply:
@genryxy if it helps to diagnose the problem, then maybe it deserves to be separate test, which is enabled.

Reply:
@olegmoz it rather checks that the test is written correctly, but does not help to diagnose the problem

Reply:
@genryxy still looks like it should be moved to another test. So we will have test that checks Artipie server may be used as target for `npm install` command and separate test to check that HTTP endpoint works properly. These are different features basically